azkaban-common/src/main/java/azkaban/imagemgmt/daos/ImageTypeDaoImpl.java

```diff
@@ -106,22 +124,82 @@ public int createImageType(final ImageType imageType) {
-      // TODO: Find a better way to get the error message. Currently apache common dbutils
-      // throws sql exception for all the below error scenarios and error message contains
-      // complete query as well, hence generic error message is thrown.
-      if (e.getErrorCode() == SQL_ERROR_CODE_DUPLICATE_ENTRY) {
-        errorMessage = "Reason: Duplicate key provided for one or more column(s).";
+      handleSqlException(e);
+    }
+    return imageTypeId;
+  }
+
+  @Override
+  public int addOwnerOfImageType(final ImageType imageType) {
+    final Set<ImageOwnership> currentOwners =
+        new HashSet<>(getImageTypeOwnership(imageType.getName()));
+    final Set<ImageOwnership> newOwners = new HashSet<>(imageType.getOwnerships());
+    newOwners.removeAll(currentOwners);
+    //Check if there are actually new owners added.
+    if (newOwners.isEmpty()) {
+      throw new ImageMgmtDaoException(ErrorCode.BAD_REQUEST, "Exception while updating image, "
+          + "specified no new owners");
+    }
+    int imageTypeId =
+        getImageTypeByName(imageType.getName()).map(BaseModel::getId).orElse(0);
+    final SQLTransaction<Integer> update = transOperator -> {
+      final Timestamp currentTimestamp = Timestamp.valueOf(LocalDateTime.now());
+      for (final ImageOwnership imageOwnership : newOwners) {
+        transOperator.update(INSERT_IMAGE_OWNERSHIP, imageTypeId, imageOwnership.getOwner(),
+            imageOwnership.getRole().name(), imageType.getCreatedBy(), currentTimestamp,
+            imageType.getModifiedBy(), currentTimestamp);
       }
-      if (e.getErrorCode() == SQL_ERROR_CODE_DATA_TOO_LONG) {
-        errorMessage = "Reason: Data too long for one or more column(s).";
+      transOperator.getConnection().commit();
+      return 1;
+    };
+    try {
+      this.databaseOperator.transaction(update);
+      if (imageTypeId < 1) {
+        log.error(String.format("Exception while updating image type due to invalid "
+            + "imageTypeId: %d.", imageTypeId));
+        throw new ImageMgmtDaoException(ErrorCode.BAD_REQUEST, "Exception while updating image ");
       }
-      throw new ImageMgmtDaoException(ErrorCode.BAD_REQUEST, "Exception occurred while creating "
-          + "image type metadata. " + errorMessage);
+      log.info("Successfully Updated image ownerships for :" + imageType.getName());
+    } catch (final SQLException e) {
+      log.error("Unable to update the image type metadata", e);
+      handleSqlException(e);
     }
     return imageTypeId;
   }
 
+  @Override
+  public int removeOwnerOfImageType(ImageType imageType) throws ImageMgmtException {
+    final Set<ImageOwnership> currentOwners =
+        new HashSet<>(getImageTypeOwnership(imageType.getName()));
+    final Set<ImageOwnership> ownersToRemove = new HashSet<>(imageType.getOwnerships());
+    ownersToRemove.retainAll(currentOwners);
+      //Check if it's removing all owners
+    if (ownersToRemove.size() >= currentOwners.size() - 1){
```

Comment:
Is this actually checking if there are two members? It seems like it is doing a size comparison.

Reply:
I can do that. I shall change it to get the final owner list size and see if it's greater than 1.